### ISSUE TITLE
Blocks sometimes incorrectly display as furnaces - possibly snow related?

### ISSUE THREAD

**Issue by Tydium** [GeyserMC/Geyser#1900]
**Describe the bug**
Basically if your in a snow biome and you break the snow. Sometimes under it will be an incorrectly displayed block (Usually a furnace)

**To Reproduce**

1. Go to a snow biome or high up in the mountains ( I was in mountains)
2. Either build a house or something and wait for it to snow.
3. break the snow off the blocks and you may just see a furnace weirdly appear in the block.
**Expected behavior**
for there not to be a weirdly placed block under the snow.
**Screenshots / Videos**
None

**Server Version and Plugins**
Using latest version of Geyser Bungeecord (Geyser version 1.2.0-SNAPSHOT (git-master-c26a2ba) (Java: 1.16.4 - 1.16.5, Bedrock: 1.16.100/1.16.101 - 1.16.200/1.16.201))

Paper 1.16.5

**Geyser Dump**

https://dump.geysermc.org/Q5iszWXwRGDBob8RjoHLHJYoRTrbfo8x

**Minecraft: Bedrock Edition Version**
Bedrock Editon on the Nintendo Switch/Windows10/and on Android (havent tested any other devices) on version 1.16.201
**Additional Context**

We noticed this happening in a 4 long 4 wide and 3 block tall house.

**Comment by Camotoy** [comment]
Please attach a list of Paper plugins on the affected server.

**Comment by Tydium** [comment]
On Bungeecord:
Pinger, reconnect_yaml, LuckPerms, SkinsRestorer, floodgate-bungee, BungeeGuard, BungeePluginManager, cmd_list, Geyser-BungeeCord, cmd_find, BungeeHub, AdvancedBan, cmd_server, UltraStaffChat, cmd_alert, cmd_send

On the 1.16.5 Backend Server (The one we completed start to finish)
 BungeeGuard, Essentials, EssentialsChat, floodgate-bukkit, Floodgate-Skript, LuckPerms, PixelPolice, PlugMan, SkinsRestorer, Skript, Vault, ViaVersion
 
 The issue im assuming is not made with plugins but just incase here they are. If you try to redownload these and try it for yourself, you will not find PixelPolice. That is a custom logging plugin made by one of my friends.

**Comment by Tydium** [comment]
CAMOTOY. I THINK I FIGURED OUT THE PROBLEM. Still unfixable by me. You know when you open a chest, crafting table, FURNACE, or anything inventory related..... That is what is causing it. I am almost certain of it.

**Comment by Camotoy** [comment]
That shouldn't trigger when you're not in an inventory, and it should only trigger for your own player.

**Comment by Tydium** [comment]
Correct. Exactly my point. When the furnace places above your head it overwrites on the client side. Obviously other players dont see it. (yet I felt this was the case because multiple of us kept using the same furnace.) the server on the other hand still thinks it was the block it was previously before (ex. dirt block)

**Comment by Tydium** [comment]
I believe there was also an issue possibly related to this. where when a block would appear above you (as should) but wouldnt dissapear. https://streamable.com/wfnhgq

**Comment by RednedEpic** [comment]
Does this still occur?
